### PR TITLE
Website 139 add indication in site search about departments news events and exhibits

### PR DESCRIPTION
**Pull request by SalazarJosh**
# Overview
This PR adds an indication to site search results in the form of a badge for Departments, News, Events, and Exhibits.

[From the planning notes document](https://docs.google.com/document/d/1iuscCMiq-tfUv7e3efOXxa7C3l73xfCsrVGU5JFeJJA/edit?tab=t.cefe4jwvhgx3#heading=h.qhm3owcn79op):

Department

- If Content Type/Page Template is Department > display badge

News

- If Content Type/Page Template Page is News > display badge

Event and Exhibits*

- If Content Type/Page Template is Event and Exhibit and the Event Type is not Exhibit > display Event badge
- If Content Type/Page Template is Event and Exhibit and the Event Type is Exhibit > display Exhibit badge

The badge styles can be [seen here in Figma](https://www.figma.com/design/RNOWJ86DGjXYlfPBAjDvBI/F24-Website-Enhancements?node-id=149-432&t=JUZ3wCHC3h4cCNaI-4) and are as follows:

- Color: neutral-300
- Text: “Badge” style in DS - all caps bold 14px, 1.25x letter spacing
- Dot: 6px circle
- Spacing*: 8px between search result title, dot, and badge

*For the dot, I just used the ● character instead of making a custom element.

We don't want these elements to be searchable, so we don't add them to the lunr indexing.

> This pull request resolves/closes/fixes [WEBSITE-139](https://mlit.atlassian.net/jira/software/projects/WEBSITE/boards/27?selectedIssue=WEBSITE-139).

The elements in search results now have a new field named `tag` that is null by default. I use the template from the `getTemplate` function to determine if a tag should be of type `department` or `news`. I couldn't do this for events / exhibits, though, since they use the same template. In that case, I added `field_event_type` to graphQL query and check against that.

The tag element itself lives in `site-search.js` and uses the `result.tag` value to show the appropriate tag. I was previously using `result.isNews`, `result.isDepartment`, etc. but that got messy real quick.

## Additional Notes
The `gatsby-config.js` file was missed in the initial eslint cleanup so there are formatting changes to that document. I added a comment to the file to indicate where I actually made changes. I need to do a separate PR to run a more comprehensive eslint cleanup again. I usually do that as part of each PR but I'm realizing there are a lot of places that I missed so a separate PR seems appropriate.

## Testing
List instructions on how to test the pull request. Some examples:

- Make sure the PR is consistent in these browsers:
  - [x] Chrome
  - [x] Firefox
  - [ ] Safari (the assignee was not able to test the pull request in this browser)
  - [x] Edge 
- Run accessibility tests:
  - [x] WAVE
  - [x] ARC Toolkit
  - [x] axe DevTools
- _How to use the feature_.
- `npm run start` and check and search for Departments, news titles (U-M staff access to library resources), event titles (A Feast For Your Eyes: Culinary Arts From Around the World), and exhibit titles (Propositions to Progress: A Working Atlas of the Global South). To see more events and exhibits, check [the Today and Upcoming page](https://lib.umich.edu/visit-and-study/events-and-exhibits/today-and-upcoming)

### CODE REVIEW

File: gatsby-config.js
Comment (SalazarJosh) [inline gatsby-config.js:1]: Some eslint automated changes tied in here. Looks like I missed cleaning up this fine in the initial pass.

File: gatsby-config.js [hunk 1]
```diff
@@ -130,32 +130,47 @@ module.exports = {
...
-        },
-      },
-    },
+            uniqname: (node) => {
+              return (node.context ? node.context.uniqname : null);
+            },
+            title: (node) => {
+              return (node.context ? node.context.title : null);
+            },
+            tag: (node) => {
```
Comment (SalazarJosh) [inline gatsby-config.js:161]: This is the only thing added to this file. The rest are eslint fixes.

File: gatsby-config.js [hunk 2]
```diff
@@ -130,32 +130,47 @@ module.exports = {
...
-            store: true,
+            store: true
           },
           {
-            name: 'uniqname',
-            store: true,
+            name: 'tag',
+            store: true
           },
+          {
```
Comment (SalazarJosh) [inline gatsby-config.js:148]: This is the only thing added to the file. The rest are eslint fixes

File: gatsby-config.js [hunk 3]
```diff
@@ -130,32 +130,47 @@ module.exports = {
...
           },
           {
             name: 'keywords',
-            store: true,
+            store: true
           },
           {
-            name: 'uniqname',
-            store: true,
+            name: 'tag',
```
Comment (SalazarJosh) [inline gatsby-config.js:145]: This tag is the only thing added to this file. The rest should be eslint fixes.

File: src/components/site-search.js [hunk 4]
```diff
@@ -412,17 +412,34 @@ LibrarySearchScopeOption.propTypes = {
...
+          <HighlightText query={query} text={result.title} />
+        </span>
+        {result.tag && (
+          <span
+            css={{
+              color: 'var(--color-neutral-300)',
+              display: 'inline-block',
+              fontSize: '14px',
+              fontWeight: 'bold',
+              letterSpacing: '1.25px',
```
Comment (erinesullivan) [inline src/components/site-search.js:435]:
```suggestion
              fontSize: '0.875rem',
              fontWeight: 'bold',
              letterSpacing: '0.0875em',
```
I understand Figma uses pixels for the typography so the measurements are an exact match, but for the sake of accessibility it's best to stick with `rem` or `em` for text measurements.

**Review by erinesullivan — APPROVED** [review]
I see the tags in the dropdown. It's a nice touch that communicates what the result will navigate to! 👍🏻

**Pull request merged**